src/Interpreters/Context.cpp

```diff
@@ -1457,6 +1458,46 @@ bool Context::hasZooKeeper() const
     return getConfigRef().has("zookeeper");
 }
 
+std::shared_ptr<BaseInterserverCredentials> Context::getInterserverCredential()
+{
+    std::lock_guard lock(shared->interserver_io_credentials_mutex);
+    return shared->interserver_io_credentials;
+}
+
+void Context::setInterserverCredentials(std::shared_ptr<BaseInterserverCredentials> credentials)
+{
+    std::lock_guard lock(shared->interserver_io_credentials_mutex);
+
+    if (credentials == nullptr)
+        return;
```

Comment:
Looks superfluous

Reply:
Again, if it changes in the future

Reply:
I'm suggesting remove it, did not see such checks things as a common thing